samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -220,10 +223,14 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
     * @return           the map of properties for the stream
     */
   private def getSystemStreamProperties(systemName: String, streamName: String) = {
-    if (systemName == null || streamName == null) {
+    if (systemName == null) {
       Map()
     }
-    config.subset(StreamConfig.STREAM_PREFIX format(systemName, streamName), true)
+    val jsc = new JavaSystemConfig(config);
```

Comment:
👍  that's the convention. Sorry, reverted to a coding standard from a past life. I'll fix it.